packages/deployer/src/compiler.ts

```diff
@@ -116,67 +150,51 @@ export class Compiler {
     public async compileAllAsync(): Promise<void> {
         await this._createArtifactsDirIfDoesNotExistAsync();
         this._contractSourcesIfExists = await Compiler._getContractSourcesAsync(this._contractsDir);
-        const contractBaseNames = _.keys(this._contractSourcesIfExists);
-        const compiledContractPromises = _.map(contractBaseNames, async (contractBaseName: string): Promise<void> => {
-            return this._compileContractAsync(contractBaseName);
+        _.forIn(this._contractSourcesIfExists, (source, fileName) => {
+            this._contractSourceData[fileName] = Compiler._getContractSpecificSourceData(source);
         });
-        await Promise.all(compiledContractPromises);
-
+        const fileNames = this._specifiedContracts.has(ALL_CONTRACTS_IDENTIFIER)
+            ? _.keys(this._contractSourcesIfExists)
+            : Array.from(this._specifiedContracts.values());
+        _.forEach(fileNames, fileName => {
+            this._setSourceTreeHash(fileName);
+        });
+        await Promise.all(_.map(fileNames, async fileName => this._compileContractAsync(fileName)));
```

Comment:
`await Promise.all(_.map(fileNames, this._compileContractAsync));`

Reply:
We need the arrow function for binding `this` here.

Reply:
i think i prefer manually calling bind here because its more explicit what the intent is

Reply:
Hmm, the way this is currently written seems pretty understandable/standard to me. Are we explicitly binding `this` in every single case where we need to throughout our codebase (rather than using arrow functions)?